docs/book/src/topics/identities-use-cases.md

```diff
@@ -0,0 +1,68 @@
+## How to use Identities with CAPZ
+
+## CAPZ controller: 
```

Comment:
Not saying we should do this in this PR, but we should probably get as granular as the specific Azure Resource Manager rights that are required, at a minimum what role is required. A role is composed of rights and there's a bunch predefined like `contributor`. Roles and rights are also scoped by subscription, group, etc.

Eventually, I think it would be best to define our own custom roles and specific scopes to ensure that we are using the least privilege possible for each role while be as clear as possible about those rights and scopes.

Example would be:

**CAPZ controller:**
- Scope: Subscription
- Role: Contributor
  - Rights: see default Azure Subscription Contributor documentation
- Reasoning: the controller is responsible for creating resource groups and cluster resources within the group. To create a resource group within a subscription, one must have subscription contributor rights. Note, this role's scope can be reduced to Resource Group Contributor if all resource groups are created prior to cluster creation.

WDYT?

Reply:
oh yeah, that is a good addition, let me try to do that

Reply:
updated